pyproject.toml

```diff
@@ -0,0 +1,115 @@
+[project]
+name = "azure-functions-worker"
+version = "4.29.0"
+description = "Python Language Worker for Azure Functions Runtime"
+authors = [
+    { name = "Azure Functions team at Microsoft Corp.", email = "azurefunctions@microsoft.com" }
+]
+keywords = ["azure", "functions", "azurefunctions",
+            "python", "serverless"]
+license = { name = "MIT", file = "LICENSE" }
+readme = { file = "README.md", content-type = "text/markdown" }
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: MacOS :: MacOS X",
+    "Environment :: Web Environment",
+    "License :: OSI Approved :: MIT License",
+    "Intended Audience :: Developers"
+]
+
+dependencies = [
+    "azure-functions==1.20.0",
+    "python-dateutil~=2.9.0"
+]
+
+# Python version-specific dependencies
+[project.optional-dependencies]
+python37 = [
+    "protobuf~=3.19.3",
+    "grpcio-tools~=1.43.0",
+    "grpcio~=1.43.0"
+]
+python38_plus = [
+    "protobuf~=4.22.0",
+    "grpcio-tools~=1.54.2",
+    "grpcio~=1.54.2",
+    "azurefunctions-extensions-base"
```

Comment:
The python37 and python38_plus should be required dependencies. We can specify what to install based on the python version like so:

"protobuf~=3.19.3; python_version == '3.7'",
"protobuf~=4.22.0; python_version >= '3.8'"

"django>2.1; os_name != 'nt'"